CMakeLists.txt

```diff
@@ -82,10 +86,11 @@ if(MPI)
 else()
    set(MPI_Fortran_LIBRARIES "")
    message("-- Not using MPI")
+   # Create an empty target with the same name
+   add_library(MPI::MPI_Fortran INTERFACE IMPORTED)
```

Comment:
You are shooting yourself in the foot here. You will only understand when you get to the packaging **and** you have some packaging tests.

Reply:
I was getting compilation failure for serial code without this line. What is the correct thing to do?

Reply:
What compilation error. Try to remove it and show it to me in the CI (you can link specific lines of the CI log)